File: document/documentation/reference/operation/local/uniform/example-1/uniform.cpp
#include "lue/framework/api/cxx.hpp"
#include "lue/document.hpp"
#include "lue/framework.hpp"
#include <hpx/hpx_main.hpp>


class Example: public lue::document::Example
{

    public:

        Example(int const argc, char const* const* argv):

            lue::document::Example{
                [&]() -> cxxopts::Options
                {
                    cxxopts::Options options{std::filesystem::path{argv[0]}.filename()};

                    options.positional_help("<result>");
                    options.add_options()
                        // clang-format off
                        ("result_array", "array to write", cxxopts::value<std::string>())
                        // clang-format on
                        ;
                    options.parse_positional({"result_array"});

                    return options;
                }(),
                argc,
                argv}

        {
        }

    private:

        auto run_implementation() -> int override
        {
            // Command line arguments
            if (!argument_parsed("result_array"))
            {
                throw std::runtime_error("Wrong usage, pass '--help' for more information");
            }

            auto const result_array_pathname = argument<std::string>("result_array");

            using namespace lue;
            using namespace lue::api;

            // [example
            using Element = LargestUnsignedIntegralElement;

            Element const min_value{1};
            Element const max_value{10};
            Field const result = uniform({6, 4}, min_value, max_value);

            to_gdal(result, result_array_pathname);
            // example]

            return EXIT_SUCCESS;
        }
};


auto main(int argc, char** argv) -> int
{
    return Example{argc, argv}.run();
}

File: document/documentation/reference/operation/local/uniform/example-1/uniform.py
import sys
from pathlib import Path

import docopt
import lue.framework_x as lfrx
from lue import __version__ as version


@lfrx.runtime_scope
def main() -> None:
    command = Path(sys.argv[0]).name
    usage = f"""\
Usage:
    {command} <result>

Options:
    -h --help   Show this screen and exit
    --version   Show version and exit
    <result>    Array to write
"""
    arguments = docopt.docopt(usage, sys.argv[1:], version=version)
    result_array_path = Path(arguments["<result>"])

    # [example
    dtype = lfrx.signed_integral_element_types[-1]
    result = lfrx.uniform(
        (6, 4),
        min_value=lfrx.as_field(lfrx.create_literal(1, dtype)),
        max_value=lfrx.as_field(lfrx.create_literal(10, dtype)),
    )

    lfrx.to_gdal(result, str(result_array_path))
    # example]


if __name__ == "__main__":
    sys.exit(main())
